dask_planner/src/parser.rs

```diff
@@ -18,6 +22,189 @@ macro_rules! parser_err {
+        })
+    }
+
+    #[pyo3(name = "getKwargs")]
+    pub fn get_kwargs(&self) -> PyResult<Vec<(String, PySqlArg)>> {
+        Ok(match &self.custom {
+            Some(CustomExpr::Nested(kwargs)) => kwargs.clone(),
+            _ => vec![],
+        })
+    }
+
+    #[pyo3(name = "getSqlType")]
+    pub fn get_sql_type(&self) -> PyResult<SqlTypeName> {
+        Ok(match &self.custom {
+            Some(custom_expr) => match custom_expr {
+                CustomExpr::Map(_) => SqlTypeName::MAP,
+                CustomExpr::Multiset(_) => SqlTypeName::MULTISET,
+                CustomExpr::Nested(_) => {
+                    return Err(py_type_err("Expected Map or Multiset, got Nested"))
+                }
+            },
+            None => match &self.expr {
+                Some(Expr::Array(_)) => SqlTypeName::ARRAY,
+                Some(Expr::Identifier(Ident { .. })) => SqlTypeName::VARCHAR,
+                Some(Expr::Value(scalar)) => match scalar {
+                    Value::Boolean(_) => SqlTypeName::BOOLEAN,
+                    Value::Number(_, false) => SqlTypeName::BIGINT,
+                    Value::SingleQuotedString(_) => SqlTypeName::VARCHAR,
+                    unexpected => {
+                        return Err(py_type_err(format!(
+                            "Expected string, got {:?}",
+                            unexpected
+                        )))
+                    }
+                },
+                Some(Expr::UnaryOp {
+                    op: UnaryOperator::Minus,
+                    expr,
+                }) => match &**expr {
+                    Expr::Value(Value::Number(_, false)) => SqlTypeName::BIGINT,
+                    unexpected => {
+                        return Err(py_type_err(format!(
+                            "Expected string, got {:?}",
+                            unexpected
+                        )))
+                    }
+                },
+                Some(unexpected) => {
+                    return Err(py_type_err(format!(
+                        "Expected array or scalar, got {:?}",
+                        unexpected
+                    )))
+                }
+                None => {
+                    return Err(py_type_err(
+                        "PySqlArg must contain either a standard or custom AST expression",
+                    ))
+                }
+            },
+        })
+    }
+
+    #[pyo3(name = "getSqlValue")]
+    pub fn get_sql_value(&self) -> PyResult<String> {
+        Ok(match &self.custom {
+            None => match &self.expr {
+                Some(Expr::Identifier(Ident { value, .. })) => value.to_string(),
+                Some(Expr::Value(scalar)) => match scalar {
+                    Value::Boolean(value) => {
+                        if *value {
+                            "1".to_string()
+                        } else {
+                            "".to_string()
+                        }
+                    }
```

Comment:
nit: we can make this more concise 

```suggestion
                    Value::Boolean(true) => "1".to_string(),
                    Value::Boolean(false) => "".to_string(),

```